zookeeper-server/src/main/java/org/apache/zookeeper/cli/SyncCommand.java

```diff
@@ -55,14 +57,18 @@ public CliCommand parse(String[] cmdArgs) throws CliParseException {
     @Override
     public boolean exec() throws CliException {
         String path = args[1];
+        CountDownLatch latch = new CountDownLatch(1);
+        final int[] resultCode = new int[1];
         try {
             zk.sync(path, new AsyncCallback.VoidCallback() {
-
                 public void processResult(int rc, String path, Object ctx) {
-                    out.println("Sync returned " + rc);
+                    resultCode[0] = rc;
+                    latch.countDown();
                 }
             }, null);
-        } catch (IllegalArgumentException ex) {
+            latch.await();
+            out.println("Sync returned " + resultCode[0]);
+        } catch (IllegalArgumentException | InterruptedException ex) {
```

Comment:
Do you need to catch `InterruptedException` here?

Reply:
`latch.await` needs this, my sweet IDEA help me do it.

Reply:
I don't it's handled properly here. You basically swallowing and hiding it inside a `MalformedPathException` which is wrong.